filebeat/input/journald/pkg/journalfield/conv_expand_test.go

```diff
@@ -121,47 +123,7 @@ var expandCapabilitiesTests = []struct {
 				"thread": mapstr.M{
 					"capabilities": mapstr.M{
 						"effective": []string{
-							"CAP_CHOWN",
-							"CAP_DAC_OVERRIDE",
-							"CAP_DAC_READ_SEARCH",
-							"CAP_FOWNER",
-							"CAP_FSETID",
-							"CAP_KILL",
-							"CAP_SETGID",
-							"CAP_SETUID",
-							"CAP_SETPCAP",
-							"CAP_LINUX_IMMUTABLE",
-							"CAP_NET_BIND_SERVICE",
-							"CAP_NET_BROADCAST",
-							"CAP_NET_ADMIN",
-							"CAP_NET_RAW",
-							"CAP_IPC_LOCK",
-							"CAP_IPC_OWNER",
-							"CAP_SYS_MODULE",
-							"CAP_SYS_RAWIO",
-							"CAP_SYS_CHROOT",
-							"CAP_SYS_PTRACE",
-							"CAP_SYS_PACCT",
-							"CAP_SYS_ADMIN",
-							"CAP_SYS_BOOT",
-							"CAP_SYS_NICE",
-							"CAP_SYS_RESOURCE",
-							"CAP_SYS_TIME",
-							"CAP_SYS_TTY_CONFIG",
-							"CAP_MKNOD",
-							"CAP_LEASE",
-							"CAP_AUDIT_WRITE",
-							"CAP_AUDIT_CONTROL",
-							"CAP_SETFCAP",
-							"CAP_MAC_OVERRIDE",
-							"CAP_MAC_ADMIN",
-							"CAP_SYSLOG",
-							"CAP_WAKE_ALARM",
-							"CAP_BLOCK_SUSPEND",
-							"CAP_AUDIT_READ",
-							"CAP_PERFMON",
-							"CAP_BPF",
-							"CAP_CHECKPOINT_RESTORE",
+							"CAP_ALL",
```

Comment:
Indeed, perhaps we should convey this better to the people managing ECS. I think this would be fine since even though they would reflect different things, they would still "semantically" reflect the same as in this process could do all it wanted.

Reply:
I'm not convince by that. If there were a flag that said ALL, then I would agree, but the ALLness is determined by comparison to the set of valid flags that were present at the time of the writing of the document. Essentially, the semantics are inferred contextually and then rendered concrete without reference to the context. I don't think this is safe. Note that ECS doesn't mention CAP_ALL explicitly, and certainly doesn't recommend that the set be collapsed to it in the case that all capabilities are present, so I don't see that we HAVE TO do this.

Reply:
To expand on this, we COULD do this if we also keep all the original flags as well. This enriches the information in the document by saying, "this is the set of capabilities that are present *and that was the complete set available at the time*".

Reply:
> I'm not convince by that. If there were a flag that said ALL, then I would agree, but the ALLness is determined by **comparison to the set of valid flags that were present at the time of the writing of the document.** Essentially, the semantics are inferred contextually and then rendered concrete without reference to the context. I don't think this is safe. Note that ECS doesn't mention CAP_ALL explicitly, and certainly doesn't recommend that the set be collapsed to it in the case that all capabilities are present, so I don't see that we HAVE TO do this.

I don't think that is quite the case, the comparison of the valid flags **are not the ones present at the time of writing the document**, they are what the kernel determines to be "all the capabilities it supports", cap.MaxBits() is discovered at [runtime](https://git.kernel.org/pub/scm/libs/libcap/libcap.git/tree/cap/cap.go#n276).
"MaxBits returns the number of kernel-named capabilities discovered at runtime in the current system."
So if the kernel adds another capability, even if we don't update the `cap` pkg, MaxBits will be bumped and CAP_ALL will reflect all the capabilities the new kernel supports.

If we don't update `cap`, the only thing we miss is the name of the new capability (which will be reflected as CAP_NUMBER in the case of not CAP_ALL).

Regarding it not being documented in ECS, this has just been suggested [here](https://github.com/elastic/security-team/issues/4375#issuecomment-1846197475), so I assume documentation will follow.

Regarding keeping both sets, the whole point is to shrink the amount of flags, since every root process will spill the 41 capabilities now.

Reply:
Answering myself, the runtime discovery _does_ what we want, it looks for the first bit that returns EINVAL, not the first bit that is not set. Tested on docker with a smaller bounding set.
```
capget({version=_LINUX_CAPABILITY_VERSION_3, pid=0}, NULL) = 0
prctl(PR_CAPBSET_READ, CAP_MAC_OVERRIDE) = 0
prctl(PR_CAPBSET_READ, 0x30 /* CAP_??? */) = -1 EINVAL (Invalid argument)
prctl(PR_CAPBSET_READ, CAP_CHECKPOINT_RESTORE) = 0
prctl(PR_CAPBSET_READ, 0x2c /* CAP_??? */) = -1 EINVAL (Invalid argument)
prctl(PR_CAPBSET_READ, 0x2a /* CAP_??? */) = -1 EINVAL (Invalid argument)
prctl(PR_CAPBSET_READ, 0x29 /* CAP_??? */) = -1 EINVAL (Invalid argument)
write(1, "maxbits=41\n", 11maxbits=41
```
and `cat /proc/self/status` is:
```
CapInh: 0000000000000000
CapPrm: 00000000a80425fb
CapEff: 00000000a80425fb
CapBnd: 00000000a80425fb
CapAmb: 0000000000000000
```

In other words, MaxBits is correct regardless of the bounding set (which containers run with a limited one), meaning we can trust that "CAP_ALL is every capability the host can have".